google/cloud/aiplatform/metadata/metadata.py

```diff
@@ -208,173 +431,221 @@ def get_experiment_df(
+                If not provided, the latest
+                step amongst all time series metrics already logged will be used.
+            wall_time (timestamp_pb2.Timestamp):
+                Optional. Wall clock timestamp when this data point is
+                generated by the end user.
 
-        if context.schema_title != expected_schema:
-            raise ValueError(
-                f"Please provide a valid {source} name. {name} is not a {source}."
-            )
-        return context.resource_name
+                If not provided, this will be generated based on the value from time.time()
 
-    def _query_runs_to_data_frame(
-        self, context_id: str, context_resource_name: str, source: str
-    ) -> "pd.DataFrame":  # noqa: F821
-        """Get metrics and parameters associated with a given Context into a Dataframe.
+        Raises:
+            RuntimeError: If current experiment run doesn't have a backing Tensorboard resource.
+        """
+        self._validate_experiment_and_run(method_name="log_time_series_metrics")
+        self._experiment_run.log_time_series_metrics(
+            metrics=metrics, step=step, wall_time=wall_time
+        )
 
+    def start_execution(
+        self,
+        *,
+        schema_title: Optional[str] = None,
+        display_name: Optional[str] = None,
+        resource_id: Optional[str] = None,
+        metadata: Optional[Dict[str, Any]] = None,
+        schema_version: Optional[str] = None,
+        description: Optional[str] = None,
+        resume: bool = False,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ) -> execution.Execution:
+        """
+        Create and starts a new Metadata Execution or resumes a previously created Execution.
+
+        To start a new execution:
+
+        ```
+        with aiplatform.start_execution(schema_title='system.ContainerExecution', display_name='trainer) as exc:
+          exc.assign_input_artifacts([my_artifact])
+          model = aiplatform.Artifact.create(uri='gs://my-uri', schema_title='system.Model')
+          exc.assign_output_artifacts([model])
+        ```
+
+        To continue a previously created execution:
+        ```
+        with aiplatform.start_execution(resource_id='my-exc', resume=True) as exc:
+            ...
+        ```
         Args:
-            context_id (str):
-                Name of the Experiment or Pipeline.
-            context_resource_name (str):
-                Full resource name of the Context associated with an Experiment or Pipeline.
-            source (str):
-                Identify whether the this is an Experiment or a Pipeline.
+            schema_title (str):
+                Optional. schema_title identifies the schema title used by the Execution. Required if starting
+                a new Execution.
+            state (gca_execution.Execution.State.RUNNING):
```

Comment:
Confirming that `state` should not be passed to the function above.

Reply:
Removed. done.